commands/provider/service_deployer.go

```diff
@@ -87,17 +89,31 @@ func readDeployReply(stream coreapi.Core_DeployServiceClient, deployment chan de
 		}
 
 		var (
+			status          = message.GetStatus()
 			serviceID       = message.GetServiceID()
 			validationError = message.GetValidationError()
 		)
 
 		switch {
+		case status != nil:
```

Comment:
This one of the reason why I drop it in refactoring. Only commands package is responsible for input/output for cli. 

Please move all output printing/spinner usage to commands package to do it properly.

Reply:
I agree 👍